Support array for all string ingest processors (#53343)
elastic/elasticsearch: docs/reference/ingest/processors/gsub.asciidoc

[[gsub-processor]]
=== Gsub Processor
Converts a string field by applying a regular expression and a replacement.

> If the field is not a string, the processor will throw an exception. If the field value is an array, all members will be converted.

Reviewer: ```suggestion
If the field is an array of string, all members of the array will be converted. If any non-string values are encountered, the processor will throw an exception. 
```
